{"ast":null,"code":"// src/until.ts\nvar until = async promise => {\n  try {\n    const data = await promise().catch(error => {\n      throw error;\n    });\n    return {\n      error: null,\n      data\n    };\n  } catch (error) {\n    return {\n      error,\n      data: null\n    };\n  }\n};\nexport { until };","map":{"version":3,"mappings":";AAeO,IAAMA,QAAQ,MAInBC,WAC6C;EAC7C,IAAI;IACF,MAAMC,OAAO,MAAMD,QAAQ,EAAEE,MAAOC,SAAU;MAC5C,MAAMA;IACR,CAAC;IACD,OAAO;MAAEA,OAAO;MAAMF;IAAK;EAC7B,SAASE,OAAP;IACA,OAAO;MAAEA;MAAOF,MAAM;IAAK;EAC7B;AACF","names":["until","promise","data","catch","error"],"sources":["/Users/ditole/Desktop/projects/pets-store/node_modules/@open-draft/until/src/until.ts"],"sourcesContent":["export type AsyncTuple<\n  ErrorType extends any = Error,\n  DataType extends any = unknown,\n> =\n  | {\n      error: ErrorType\n      data: null\n    }\n  | { error: null; data: DataType }\n\n/**\n * Gracefully handles a given Promise factory.\n * @example\n * const { error, data } = await until(() => asyncAction())\n */\nexport const until = async <\n  ErrorType extends any = Error,\n  DataType extends any = unknown,\n>(\n  promise: () => Promise<DataType>,\n): Promise<AsyncTuple<ErrorType, DataType>> => {\n  try {\n    const data = await promise().catch((error) => {\n      throw error\n    })\n    return { error: null, data }\n  } catch (error) {\n    return { error, data: null }\n  }\n}\n"]},"metadata":{},"sourceType":"module","externalDependencies":[]}